src/main/java/org/veupathdb/service/eda/ds/plugin/CategoricalCountsPlugin.java

```diff
@@ -0,0 +1,140 @@
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static org.gusdb.fgputil.FormatUtil.TAB;
+
+public class CategoricalCountsPlugin extends AbstractEmptyComputePlugin<CategoricalCountsPostRequest, CategoricalCountsSpec> {
+  
+  private static final Logger LOG = LogManager.getLogger(CategoricalCountsPlugin.class);
+
+  private static final ManagedMap<String,CategoricalCountsPostResponse> RESULT_CACHE =
+      new ManagedMap<>(5000, 2000);
+
+  private String _cacheKey;
+  private CategoricalCountsPostResponse _cachedResponse;
+
+  @Override
+  protected Class<CategoricalCountsPostRequest> getVisualizationRequestClass() {
+    return CategoricalCountsPostRequest.class;
+  }
+
+  @Override
+  protected Class<CategoricalCountsSpec> getVisualizationSpecClass() {
+    return CategoricalCountsSpec.class;
+  }
+
+  @Override
+  public ConstraintSpec getConstraintSpec() {
+    return new ConstraintSpec()
+      .dependencyOrder(List.of("variable"))
+      .pattern()
+        .element("variable")
+          .shapes(APIVariableDataShape.CATEGORICAL, APIVariableDataShape.ORDINAL)
+          .description("Variable must be categorical.")
+      .done();
+  }
+  
+  @Override
+  protected void validateVisualizationSpec(CategoricalCountsSpec pluginSpec) throws ValidationException {
+    validateInputs(new DataElementSet()
+      .var("variable", pluginSpec.getVariable()));
+  }
+
+  @Override
+  protected List<StreamSpec> getRequestedStreams(CategoricalCountsSpec pluginSpec) {
+    String entityId = pluginSpec.getVariable().getEntityId();
+    _cacheKey = EncryptionUtil.md5(
+        JsonUtil.serializeObject(getSubsetFilters()) +
+        "|" + JsonUtil.serializeObject(pluginSpec)
+    );
+    _cachedResponse = RESULT_CACHE.get(_cacheKey);
+    if (_cachedResponse != null) LOG.info("Found cached result.");
+
+    return _cachedResponse != null ? Collections.emptyList() : ListBuilder.asList(
+      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getVariable().getEntityId())
+        .addVar(pluginSpec.getVariable()));
+  }
+
+  @Override
+  protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
+    CategoricalCountsSpec spec = getPluginSpec();
+
+    // create scanner and line parser
+    InputStreamReader isReader = new InputStreamReader(new BufferedInputStream(dataStreams.get(DEFAULT_SINGLE_STREAM_NAME)));
+    BufferedReader reader = new BufferedReader(isReader);
+    DelimitedDataParser parser = new DelimitedDataParser(reader.readLine(), TAB, true);
+
+    // establish column header indexes
+    Function<VariableSpec,Integer> indexOf = var -> parser.indexOfColumn(getUtil().toColNameOrEmpty(var)).orElseThrow();
```

Comment:
This seems like overkill here since `indexOf` is only used once.  Why not just call parser.indexOfColumn with spec.getVariable()?